### PR TITLE
Devices Tray keep itself updated

### PR DESCRIPTION
**Pull request by kalkin**
* Ported to the new D-Bus API
* Reacts to Device Added/Removed & Attached/Detached Signals

### CODE REVIEW

File: qui/models/qubes.py [hunk 1]
```diff
@@ -47,93 +46,156 @@ def __call__(cls, *args, **kwargs):
...
+    ''' Wrapper around `org.qubes.Device` Interface '''
 
     def __init__(self, proxy: dbus.proxies.ProxyObject,  # pylint: disable=no-member
                  data: dbus.Dictionary=None) -> None:  # pylint: disable=no-member
         super().__init__(proxy)
+        # TODO Add handling for usb & mic when they are ready
+        if self._data['dev_class'] == 'block':
+            self.icon = 'drive-removable-media'
+        else:
+            self.icon = 'network-wired-symbolic'
```
Comment (marmarek) [inline qui/models/qubes.py]: `device_hbox()` alread handle icons.

**Comment by kalkin** [comment]
@marmarek Fixed

**Comment by marmarek** [comment]
I get this:
```
Traceback (most recent call last):
  File "/usr/bin/qui-devices", line 9, in <module>
    load_entry_point('qui==0.1', 'gui_scripts', 'qui-devices')()
  File "/usr/lib/python3.5/site-packages/pkg_resources/__init__.py", line 542, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2575, in load_entry_point
    return ep.load()
  File "/usr/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2235, in load
    return self.resolve()
  File "/usr/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2241, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/lib/python3.5/site-packages/qui/tray/devices.py", line 22, in <module>
    DEVICES = qui.models.qubes.DevicesManager()
  File "/usr/lib/python3.5/site-packages/qui/models/qubes.py", line 114, in __init__
    super().__init__(proxy, cls=Device)
  File "/usr/lib/python3.5/site-packages/qui/models/dbus.py", line 176, in __init__
    self.children[child_proxy.object_path] = cls(child_proxy, _data)
TypeError: Can't instantiate abstract class Device with abstract methods _setup_signals
```

**Comment by kalkin** [comment]
Sorry should befixed

**Comment by marmarek** [comment]
It's better, but now I get this:
```
Traceback (most recent call last):
  File "/usr/bin/qui-devices", line 9, in <module>
    load_entry_point('qui==0.1', 'gui_scripts', 'qui-devices')()
  File "/usr/lib/python3.5/site-packages/qui/tray/devices.py", line 304, in main
    app.run()
  File "/usr/lib/python3.5/site-packages/qui/tray/devices.py", line 294, in run
    self.devices.add(obj_path)
  File "/usr/lib/python3.5/site-packages/qui/tray/devices.py", line 216, in add
    self._insert(dev_obj_path, position)
  File "/usr/lib/python3.5/site-packages/qui/tray/devices.py", line 231, in _insert
    menu_item = DeviceItem(dev_obj_path)
  File "/usr/lib/python3.5/site-packages/qui/tray/devices.py", line 176, in __init__
    label_path = self.dev.backend_domain['label']  # type: dbus.ObjectPath
  File "/usr/lib/python3.5/site-packages/qui/models/qubes.py", line 90, in backend_domain
    return DomainManager().children[vm_obj_path]
KeyError: dbus.ObjectPath('/org/qubes/DomainManager1/domains/12', variant_level=1)
```
No idea what is domain 12:
```
$ qvm-ls -O name,qid
NAME           QID
debian-8       1
dom0           0
fedora-25      2
fedora-25-dvm  10
personal       7
sys-firewall   4
sys-net        3
sys-usb        8
testvm         11
untrusted      5
vault          6
work           9
```

Maybe some caching? Killing all dbus services first fixes this (I think...). Now I get this:
```
Traceback (most recent call last):
  File "/usr/bin/qui-devices", line 9, in <module>
    load_entry_point('qui==0.1', 'gui_scripts', 'qui-devices')()
  File "/usr/lib/python3.5/site-packages/qui/tray/devices.py", line 304, in main
    app.run()
  File "/usr/lib/python3.5/site-packages/qui/tray/devices.py", line 294, in run
    self.devices.add(obj_path)
  File "/usr/lib/python3.5/site-packages/qui/tray/devices.py", line 216, in add
    self._insert(dev_obj_path, position)
  File "/usr/lib/python3.5/site-packages/qui/tray/devices.py", line 231, in _insert
    menu_item = DeviceItem(dev_obj_path)
  File "/usr/lib/python3.5/site-packages/qui/tray/devices.py", line 178, in __init__
    hbox = qui.decorators.device_hbox(self.dev)  # type: Gtk.Box
  File "/usr/lib/python3.5/site-packages/qui/decorators.py", line 86, in device_hbox
    dev_icon = create_icon(icon)
  File "/usr/lib/python3.5/site-packages/qui/decorators.py", line 116, in create_icon
    icon_dev = Gtk.IconTheme.get_default().load_icon(name, 16, 0)
GLib.Error: gtk-icon-theme-error-quark: Icon 'generic-usb' not present in theme Adwaita (0)
```
But the icon is there: `/usr/share/icons/Adwaita/22x22/devices/generic-usb.png`. Maybe it's about size (I see `16` in the traceback)? If that's the case, the icon should be downscalled to all possible sizes there (for example using convert, at build time). What is original size?

**Comment by kalkin** [comment]
Can you try running `gtk-update-cache`? I think we need to add it to the spec file

**Comment by marmarek** [comment]
After running `gtk-update-icon-cache /usr/share/icons/Adwaita` it's better. I'll add it to spec.

**Pull request merged**